### PR TITLE
feat(add proxy): add proxy for upgrade

### PR DESCRIPTION
**Pull request by sinhaashish**
Signed-off-by: sinhaashish <ashi.sinha.87@gmail.com>

### CODE REVIEW

File: k8s/proxy/src/proxy.rs [hunk 1]
```diff
@@ -312,3 +341,83 @@ impl ConfigBuilder<Loki> {
...
+    /// This is simply a boxed `tower::Service` so can be used for any HTTP requests.
+    pub async fn build(self) -> anyhow::Result<(Uri, UpgradeOperatorClient)> {
+        match self.method {
+            ForwardingProxy::HTTP => self.build_http().await,
+            ForwardingProxy::TCP => self.build_tcp().await,
+        }
+    }
+    /// Tries to build an HTTP `Configuration` from the current self.
+    async fn build_http(self) -> anyhow::Result<(Uri, UpgradeOperatorClient)> {
+        println!("port forward being done here");
```
Comment (tiagolobocastro) [inline k8s/proxy/src/proxy.rs]: please remove this

File: k8s/proxy/src/proxy.rs [hunk 2]
```diff
@@ -312,3 +341,83 @@ impl ConfigBuilder<Loki> {
...
+        let uri = kube_forward::HttpForward::new(self.target, Some(self.scheme.into()))
+            .await?
+            .uri()
+            .await?;
+        let config = super::config_from_kubeconfig(self.kube_config).await?;
+        let client = kube::Client::try_from(config)?;
+        let proxy = kube_forward::HttpProxy::new(client);
+
+        println!(" what is the url {:?}", uri);
```
Comment (tiagolobocastro) [inline k8s/proxy/src/proxy.rs]: please remove this

**Review by tiagolobocastro — APPROVED** [review]

File: k8s/proxy/src/proxy.rs (hunk 1 above)
Comment (sinhaashish) [reply k8s/proxy/src/proxy.rs]: soory , my bad

**Comment by sinhaashish** [comment]
bors try

**Review by Abhinandan-Purkait — APPROVED** [review]

**Comment by Abhinandan-Purkait** [comment]
@sinhaashish  You would probably need to fix the test script to point to the correct plugin package as you have changed the name of the package.

```

+ cargo test -p kubectl-plugin -- --test-threads=1

error: package ID specification `kubectl-plugin` did not match any packages

++ cleanup_handler

**Comment by sinhaashish** [comment]
bors try

**Comment by sinhaashish** [comment]
bors merge

**Comment by sinhaashish** [comment]
bors try

**Comment by sinhaashish** [comment]
bors try

**Comment by sinhaashish** [comment]
bors merge

**Pull request merged**